btf/dedup.go

```diff
@@ -0,0 +1,457 @@
+package btf
+
+import (
+	"fmt"
+	"hash/maphash"
+	"slices"
+)
+
+func Deduplicate(roots []Type) ([]Type, error) {
```

Comment:
Please add a doc comment.

Reply:
I think we can get rid of it, minimize the API surface. Its was useful while testings, but we can just call the unexported function to do that.